js/components/grid/calendar.js.jsx

```diff
@@ -1,301 +1,285 @@
-import React from "react"
+import { createRef, useCallback } from "react"
 import { MapModal } from "../common/react_modal.js.jsx"
 
 /*
  * Holds the containers of the Fall and Spring timetables,
  * and performs some pre-processing steps with a list of 'Lecture' objects
  */
-export class Row extends React.Component {
-  render() {
-    // Create a list of lecture objects
-    let lectures = this.props.lectureSections.map(
-      (lectureSection, index, lectureSections) =>
-        createNewLectures(lectureSection, index, lectureSections)
-    )
-    lectures = [].concat.apply([], lectures)
-
-    const fallLectures = lectures.filter(lecture => {
-      return lecture.session === "F" || lecture.session === "Y"
-    })
-
-    const springLectures = lectures.filter(lecture => {
-      return lecture.session === "S" || lecture.session === "Y"
-    })
-
-    // Organize the structure of the <fallSession> and <springSession> 2-D dictionaries
-    let fallSession, springSession
-    ;[fallSession, springSession] = initializeSessions(lectures)
-
-    // For each session, set the 'width' attribute of each Lecture
-    // (go to Lecture constructor for definition of 'width')
-    setWidths(fallSession)
-    setWidths(springSession)
-
-    // Fill session colSpans dictionaries
-    const fallColSpans = { M: 0, T: 0, W: 0, R: 0, F: 0 }
-    const springColSpans = { M: 0, T: 0, W: 0, R: 0, F: 0 }
-    storeColSpans(fallSession, fallColSpans)
-    storeColSpans(springSession, springColSpans)
-    // Generate a container for each of the Fall and Spring timetables individually
-    return (
-      <>
-        <TimetableContainer
-          session="F"
-          lecturesByTime={fallSession}
-          headColSpans={fallColSpans}
-          lectures={fallLectures}
-        />
-        <TimetableContainer
-          session="S"
-          lecturesByTime={springSession}
-          headColSpans={springColSpans}
-          lectures={springLectures}
-        />
-      </>
-    )
-  }
+export const Row = props => {
```

Comment:
You should be able to use the standard function syntax here and throughout, `export function Row(props) { ... }`